### PR TITLE
Yeezick/native applicant owner review flow

### PR DESCRIPTION
**Pull request by yeezick**
This PR accomplishes (BC-108: Project Owner Applicant Flow)[https://bootcamper.atlassian.net/jira/software/c/projects/BC/boards/2?modal=detail&selectedIssue=BC-108] , enabling project owners to add or reject applicants to their team. 
- `OwnerOptions` is rendered conditionally based on params 
- Initial render of an applicant's profile will check if applicant is still in `interested_applicants` of project passed thru params to handle whether or not trigger the "review sequence"
- Owner's decision affects the contents of the `DecisionModal` 

CONCERNS:
- Backend responded with 200 even though the requests were failing to update the project and user models (was using `projectId` instead of `projectID`) ; fixed now, but error handling needs improvement. 
- I did not have the chance to thoroughly test this before sprint close

### CODE REVIEW

File: native/screens/Applicants/Applicants.jsx [hunk 1]
```diff
@@ -3,45 +3,60 @@ import { useEffect, useState } from 'react';
...
 
-    setEngineers(() => filterRoles('Software Engineer'));
-    setDesigners(() => filterRoles('UX Designer'));
-  }, [reduxUserID, soloProject]);
+      setCurrProject(fetchedProject);
+      setEngineers(() => filterRoles('Software Engineer'));
+      setDesigners(() => filterRoles('UX Designer'));
+    };
+    setProject();
+  }, [route.params]);
```
Comment (gooliajulia) [inline native/screens/Applicants/Applicants.jsx:33]: Does having route.params as the second argument here take care of making sure the screen re-renders when a user is routed here?

File: native/screens/UserDashboard.jsx [hunk 2]
```diff
@@ -1,65 +1,62 @@
...
-          {myProjects.map((project)=> {
-            return (
-              <ProjectBanner 
-                title={project.title}
-                role={project.owner === userId ? "Project Owner" : "Collaborator"}
-                projectID={project._id}
-                navigation={navigation}
-              />
-            )
+        {myProjects.map((project, idx) => {
```
Comment (gooliajulia) [inline native/screens/UserDashboard.jsx:28]: curious about adding the idx here as it isn't ever used below?

File: native/screens/UserProfile.jsx [hunk 3]
```diff
@@ -16,24 +25,50 @@ export const UserProfile = ({ route, navigation }) => {
...
+        }
+      };
+      toggleReviewSequence();
+    }
+  }, [route.params]);
+
   useEffect(() => {
     const setUser = async () => {
-      if (route.params === undefined || route.params.userID === reduxUser._id) {
+      if (!route.params || route.params.userID === reduxUser._id) {
```
Comment (gooliajulia) [inline native/screens/UserProfile.jsx:60]: Nice switch here 👍

**Review by gooliajulia — APPROVED** [review]
This looks good to me 👍

File: native/screens/Applicants/Applicants.jsx (hunk 1 above)
Comment (yeezick) [reply native/screens/Applicants/Applicants.jsx:33]: Correct! Even though the `route.params` ends up being an identical object, React probably creates a new object in memory every time, forcing it to think new values have been passed. It's the only workaround i've discovered so far.

**Pull request merged**